.github/workflows/docs.yml

```diff
@@ -0,0 +1,32 @@
+name: Check Documentation
+on:
+  workflow_dispatch:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  rustdoc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        name: Checkout
+
+      - uses: actions-rs/toolchain@v1
+        name: Install rust toolchain
+        with:
+          toolchain: nightly
+          override: true
+
+      - uses: Swatinem/rust-cache@v2
+        name: Smart caching for Cargo
+
+      - uses: actions-rs/cargo@v1
+        name: Check Documentation with Rustdoc
```

Comment:
How is this doing any checks? As far as I can tell it builds the documentation from the sources but doesn't do anything with it after that.

Reply:
"Check" is probably what's misleading here. It simply checks if the documentation builds. The build artifact could also be used to generate a static website over github pages or similar if need be.